arch/arc/include/swap_macros.h

```diff
@@ -44,10 +44,17 @@ extern "C" {
 
 #ifdef CONFIG_USERSPACE
 #ifdef CONFIG_ARC_HAS_SECURE
+#ifdef CONFIG_ARC_SECURE_FIRMWARE
```

Comment:
not sure we cover all combination of options, looks like HAS_SECURE implies we also have USERSPACE - no support for Secure but no userspace?

Reply:
if no userspace, no need to save and restore  USER/KERNEL_SP realted aux regs.